var/spack/repos/builtin/packages/intel-mpi/package.py

```diff
@@ -62,7 +74,13 @@ def setup_dependent_package(self, module, dep_spec):
         else:
             raise RuntimeError('No suitable bindir found')
 
-        self.spec.mpicc = join_path(bindir, 'mpicc')
```

Comment:
I don't think it's correct. We set environment variables (`I_MPI_CC` and friends) to make sure `Intel-MPI` will use Spack's compiler wrappers regardless whether it's `gcc` compiler or `intel`. When you tell it to do `mpiicc` (`intel` only), compiler wrappers are probably out of the picture all together.

One way or another, IMO this change is redundant.

@lee218llnl ping.